### ISSUE TITLE
svn no longer works with github.com for raspberrypi/firmware

### ISSUE THREAD

**Issue by vanillaSprinkles** [antonym/netboot.xyz#1421]
**Describe the bug**
the ansible templates Makefiles for rpi3 and rpi4 currently use `svn` to pull down the raspberrypi/firmware repo, branch stable, subdir boot; github.com has since stopped supporting svn ( for reference: https://github.blog/2023-01-20-sunsetting-subversion-support/ )

https://github.com/netbootxyz/netboot.xyz/blob/development/roles/netbootxyz/templates/pipxe/Makefile-rpi3.j2#L27

https://github.com/netbootxyz/netboot.xyz/blob/development/roles/netbootxyz/templates/pipxe/Makefile-rpi4.j2#L27

**To Reproduce**
1. git clone the repo
2. set `generate_disks_rpi: true` (in `script/netbootxyz-overrides.yml` )
3. run the docker build: (`docker build -t localbuild -f Dockerfile .`)

**Expected behavior**
rpi stuff to build

**Additional context**
(edited)
sparse checkout gets the job done (among other methods) 

i put in a pull request (at https://github.com/netbootxyz/pipxe/pull/2) to address the issue - which should also deprecate the need for jinja2 files to generate the make files and instead be called with several of the make env options. (i also put a PR in for ipxe/pipxe with the same - would be nice if all these forks were centrally managed with several input options)

**Comment by Furbynat0r1122** [comment]
Do you know any workaround`?

**Comment by vanillaSprinkles** [comment]
yes i posted some pull requests in to fix it; if the single Makefile for both rpi3 and rpi4 gets merged and accepted (along with the rpi4 img working as expected) then the j2 Makefiles provided by ansible in netboot.xyz can be deprecated - as well as the playbook being addressed to handle the new output names of the img file(s)

edit: i wont be able to test rpi4 directly; and my rpi3 i plan to test eventually; i did let the github actions build img's for each on my for of the repo (using the same versions netboot.xyz has checked out for the sub-git-modules and the Makefile with options inspired by netboot's ansible work) - if you test any of them before i do, please give feedback

**Comment by antonym** [comment]
I have RPI 3,4,5s to test on, I had disable the builds because they were originally failing in the iPXE build. I'll try and find some time to test to see if I can get a successful build and if it boots. At one point iPXE would load the menus but never had success netbooting kernels so I stopped chasing it down.

**Comment by vanillaSprinkles** [comment]
@antonym  interesting; i did see another fork of ipxe/pipxe where i assume they had it functioning (found it of course after i got the thing building myself) - it was at https://github.com/valtzu/pipxe

if you didnt see my fork, i let the github actions build and made a release from https://github.com/vanillaSprinkles/fork__ipxe__pipxe/tree/vs/20240225/refactor_svn_git_update_build_os_to_ubuntu_22.04
